packages/cosmic-swingset/lib/ag-solo/vats/lib-dehydrate.js

```diff
@@ -0,0 +1,82 @@
+import harden from '@agoric/harden';
+import { makeMarshal } from '@agoric/marshal';
+import makeStore from '@agoric/store';
+import { assert, details } from '@agoric/assert';
+
+// Marshalling for the UI should only use the user's petnames. We will
+// call marshalling for the UI "dehydration" and "hydration" to distinguish it from
+// other marshalling.
+export const makeDehydrator = (initialUnnamedCount = 0) => {
+  let unnamedCount = initialUnnamedCount;
+
+  const petnameKindToMappings = makeStore();
+
+  const searchOrder = [];
+
+  const makeMappings = kind => {
+    assert.typeof(kind, 'string', details`kind ${kind} must be a string`);
+    const valToPetname = makeStore();
+    const petnameToVal = makeStore();
+    const addPetname = (petname, val) => {
+      petnameToVal.init(petname, val);
+      valToPetname.init(val, petname);
+    };
+    const mappings = harden({
```

Comment:
Calling this "mappings" I find confusing, especially since there are several of these, one per kind.

Rather, mathematically, each is a one-to-one "mapping" or "bijection". The fact that this one-to-one mapping is implemented by two stores is secondary.

Reply:
Ok, I will rename to 'mapping' (singular)